### PR TITLE
Add support for vm validation annotation

### PR DESCRIPTION
**Pull request by omeryahud**
**What this PR does / why we need it**:
In this PR, I introduce support for VM validation annotation.
This annotation is placed in order to drop the dependency of the VM on a template, that might be updated or removed during the lifetime of the VM.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
Template validator now gives precedence to the 'vm.kubevirt.io/validations' annotation over the template's 'validations' annotation when looking for validation rules.
```

### CODE REVIEW

File: functests/15-test-vm-with-validations-template-without-validations.sh [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+echo "[test_id:5034]: Template without validations, VM with validations"
+$KUBECTL create -f manifests/15-template-without-validations-vm-with-validations/template-without-rules.yaml
+
+# Create a VM that should pass validation
+echo "should create a VM that passes validation"
+$KUBECTL create -f manifests/15-template-without-validations-vm-with-validations/vm-with-validation-rules-passing.yaml && RET_A=0
+$KUBECTL delete vm vm-test-15
+
+# Attempt to create a VM that should fail validation
+echo "should fail to create VM that fails validation"
```
Comment (rnetser) [inline functests/15-test-vm-with-validations-template-without-validations.sh:14]: test id should be here - [test_id:5034]

File: functests/15-test-vm-with-validations-template-without-validations.sh [hunk 2]
```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+{
+RET_A=1
+RET_B=1
+echo "[test_id:5034]: Template without validations, VM with validations"
+$KUBECTL create -f manifests/15-template-without-validations-vm-with-validations/template-without-rules.yaml
+
+# Create a VM that should pass validation
+echo "should create a VM that passes validation"
```
Comment (rnetser) [inline functests/15-test-vm-with-validations-template-without-validations.sh:9]: test_id:5173

File: functests/16-test-vm-with-validations-template-with-validations.sh [hunk 3]
```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+{
+RET_A=1
+RET_B=1
+echo "[test_id:5035]: Template with validations, VM with validations"
+# Template expects max amount of CPU cores to be 4
+$KUBECTL create -f manifests/16-template-with-validations-vm-with-validations/template-with-rules.yaml
+
+# Creating a VM with a rule of maximum 6 cores and actual 5 cores, should pass
+echo "should successfully create a VM based on the VM validation rules"
```
Comment (rnetser) [inline functests/16-test-vm-with-validations-template-with-validations.sh:10]: test_id:5036

File: functests/16-test-vm-with-validations-template-with-validations.sh [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+$KUBECTL create -f manifests/16-template-with-validations-vm-with-validations/vm-with-validation-rules-passing.yaml && RET_A=0
+$KUBECTL delete vm vm-test-16
+
+# Creating a VM with a rule of maximum 4 cores and actual 5 cores, should fail
+echo "[test_id:5036]: Template with validations, VM with validations"
```
Comment (rnetser) [inline functests/16-test-vm-with-validations-template-with-validations.sh:15]: test_id:5035

File: functests/17-test-vm-with-validations-no-template.sh [hunk 5]
```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+{
+RET_A=1
+RET_B=1
+echo "[test_id:5046]: VM with validations and deleted template"
```
Comment (rnetser) [inline functests/17-test-vm-with-validations-no-template.sh:5]: test_id:5174

File: functests/17-test-vm-with-validations-no-template.sh [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+echo "should fail to create a VM based on the VM validation rules"
+$KUBECTL create -f manifests/17-vm-with-validations-no-template/vm-with-validation-rules-deleted-template-failing.yaml && RET_A=1
+
+echo "[test_id:5047]: VM with validations without template"
+echo "should successfully create a VM based on the VM validation rules"
+$KUBECTL create -f manifests/17-vm-with-validations-no-template/vm-with-validation-rules-without-template-passing.yaml && RET_B=0
+$KUBECTL delete vm vm-test-17
+
+echo "should fail to create a VM based on the VM validation rules"
```
Comment (rnetser) [inline functests/17-test-vm-with-validations-no-template.sh:18]: test_id:5047

File: functests/17-test-vm-with-validations-no-template.sh [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+RET_B=1
+echo "[test_id:5046]: VM with validations and deleted template"
+echo "should successfully create a VM based on the VM validation rules"
+$KUBECTL create -f manifests/17-vm-with-validations-no-template/vm-with-validation-rules-deleted-template-passing.yaml && RET_A=0
+$KUBECTL delete vm vm-test-17
+
+echo "should fail to create a VM based on the VM validation rules"
+$KUBECTL create -f manifests/17-vm-with-validations-no-template/vm-with-validation-rules-deleted-template-failing.yaml && RET_A=1
+
+echo "[test_id:5047]: VM with validations without template"
```
Comment (rnetser) [inline functests/17-test-vm-with-validations-no-template.sh:13]: test_id:5175

**Review by rnetser — COMMENTED** [review]
Updated the test_ids

**Comment by omeryahud** [comment]
@rnetser Modified the test ids

**Comment by omeryahud** [comment]
@akrejcir @ksimon1 @vatsalparekh @shwetaap 
Can I get a review on this please?

**Comment by omeryahud** [comment]
@ksimon1 @akrejcir 
Can you review again please? I had to rebase

**Pull request merged**